lib/schema/Schemas.js

```diff
@@ -0,0 +1,396 @@
+        }
+    },
+    "$schema": "http://json-schema.org/schema#",
+    "type": "object",
+    "if": {
+        "properties": {
+            "pluginType": {
+                "const": "library"
+            }
+        }
+    },
+    "then": {
+        "required": ["libraryName", "libraryVersion"]
+    },
+    "else": {
+        "if": {
+            "properties": {
+                "pluginType": {
+                    "const": "nodeAuthentication"
+                }
+            }
+        },
+        "then": {
+            "required": ["authenticationCategory", "filename"]
+        },
+        "else": {
+            "if": {
+                "properties": {
+                    "pluginType": {
+                        "const": "desktop"
+                    }
+                }
+            },
+            "then": {
+                "properties": {
+                    "webContent": {
+                        "if": {
+                            "additionalProperties": false //Checks if the webcontent is empty
+                        },
+                        "then": {},
+                        "else": {
+                            "allOf": [
+                                { "$ref": "#/definitions/webContent" },
+                                { "required": ["framework"] }
+                            ]
+                        }
+                    }
+                }
+            },
+            "else": {
+                "properties": {
+                    "webContent": {
+                        "if": {
+                            "additionalProperties": false //Checks if the webcontent is empty
+                        },
+                        "then": {},
+                        "else": {
+                            "allOf": [
+                                { "$ref": "#/definitions/webContent" },
+                                {"required": ["framework", "launchDefinition", "descriptionKey", "descriptionDefault", "defaultWindowStyle"] }
+                            ]
+                        }
+                    }
+                }
+            }
+        }
+    },
+    "properties": {
+        "pluginType": {
+            "type": "string",
+            "enum": ["application", "bootstrap", "desktop", "library", "nodeAuthentication", "windowManager"],
+            "errorMessage": {
+                "type": "must be a string",
+                "enum": "must be one of the following values: \n  application\n  bootstrap\n  desktop\n  library\n  nodeAuthentication\n  windowManager"
+            }
+        },
+        "identifier": {
+            "type": "string",
+            "errorMessage": {
+                "type": "must be a string"
+            }
+        },
+        "pluginVersion": {
+            "type": "string",
+            "pattern": semverRegex,
+            "errorMessage": {
+                "type": "must be a string",
+                "pattern": "must follow semantic versioning spec (x.y.z-anything)"
```

Comment:
Please cleanup the logs here, I'm seeing a bit of redundant info. I intentionally used the wrong version to cause a schema problem, and see this:

```
[2018-12-04 14:32:07.768 _unp.bootstrap INFO] - Validating Plugin Definition at ../../explorer-server-auth//pluginDefinition.json
[2018-12-04 14:32:07.790 _unp.bootstrap INFO] - Loaded Plugin Defintion Schema
Error: Key "pluginVersion" must follow semantic versioning spec (x.y.z-anything), is 1.0
    at Object.validatePluginDef (C:\Users\sgrady\Downloads\zlux-404\zlux-proxy-server\js\pluginDefinitionValidation.js:73:15)
    at PluginLoader._readPluginDef (C:\Users\sgrady\Downloads\zlux-404\zlux-proxy-server\js\plugin-loader.js:548:32)
    at PluginLoader.loadPlugins (C:\Users\sgrady\Downloads\zlux-404\zlux-proxy-server\js\plugin-loader.js:598:29)
    at Server.<anonymous> (C:\Users\sgrady\Downloads\zlux-404\zlux-proxy-server\js\index.js:270:23)
    at Generator.next (<anonymous>)
    at Generator.tryCatcher (C:\Users\sgrady\Downloads\zlux-404\zlux-proxy-server\js\node_modules\bluebird\js\release\util.js:16:23)
    at PromiseSpawn._promiseFulfilled (C:\Users\sgrady\Downloads\zlux-404\zlux-proxy-server\js\node_modules\bluebird\js\release\generators.js:97:49)
    at Server.start (C:\Users\sgrady\Downloads\zlux-404\zlux-proxy-server\js\node_modules\bluebird\js\release\generators.js:201:15)
    at Object.<anonymous> (C:\Users\sgrady\Downloads\zlux-404\zlux-example-server\js\zluxServer.js:18:13)
    at Module._compile (internal/modules/cjs/loader.js:688:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:699:10)
    at Module.load (internal/modules/cjs/loader.js:598:32)
    at tryModuleLoad (internal/modules/cjs/loader.js:537:12)
    at Function.Module._load (internal/modules/cjs/loader.js:529:3)
    at Function.Module.runMain (internal/modules/cjs/loader.js:741:12)
    at startup (internal/bootstrap/node.js:285:19)
    at bootstrapNodeJSCore (internal/bootstrap/node.js:739:3)
[2018-12-04 14:32:07.801 _unp.bootstrap WARNING] - Error: Key "pluginVersion" must follow semantic versioning spec (x.y.z-anything), is 1.0
[2018-12-04 14:32:07.801 _unp.bootstrap INFO] - Failed to load com.zowe.explorer.server.auth.json
```

1. Don't need the same message twice
2. Don't need a stack trace at all (You can solve 1 and 2 here but eliminating the non-_unp.bootstrap log and then this PR gets rid of that stack trace: https://github.com/zowe/zlux-proxy-server/pull/33)
3. Do need to know which plugin (by ID) failed on the same line that prints the reason for failure. I know that the info is a few lines above and below, but when people send you error messages, they do not always expect that, and therefore having the info in the same message can reduce common questions & mistakes.